### PR TITLE
Refactor cloud prepare flags docs

### LINKED ISSUE #731: Reorganize `subctl cloud prepare` documentation
Add a section for common flags for all clouds so it will not be repeated.

### LINKED ISSUE #848: Document the missing important flags in `subctl cloud prepare gcp`
From the latest `subctl cloud prepare gcp ...` we can see that there are many new flags. Although its not necessary to document every flag, some of the important ones can be identified and documented as it will help users to be aware of such a flag. 
For example: `--insecure-skip-tls-verify`

Following is the list of options that are currently supported:

```
Usage:
  subctl cloud prepare gcp [flags]

Flags:
      --credentials string        GCP credentials configuration file (default "/home/sgaddam/.gcp/osServiceAccount.json")
      --dedicated-gateway         Whether a dedicated gateway node has to be deployed (default true)
      --gateway-instance string   Type of gateway instance machine (default "n1-standard-4")
      --gateways int              Number of gateways to deploy (default 1)
  -h, --help                      help for gcp
      --infra-id string           GCP infra ID
      --ocp-metadata string       OCP metadata.json file (or the directory containing it) from which to read the GCP infra ID and region from (takes precedence over the specific flags)
      --project-id string         GCP project ID
      --region string             GCP region

Global Flags:
      --as string                      Username to impersonate for the operation
      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --as-uid string                  UID to impersonate for the operation
      --certificate-authority string   Path to a cert file for the certificate authority
      --client-certificate string      Path to a client certificate file for TLS
      --client-key string              Path to a client key file for TLS
      --cluster string                 The name of the kubeconfig cluster to use
      --context string                 The name of the kubeconfig context to use
      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string              absolute path(s) to the kubeconfig file(s)
      --nat-discovery-port uint16      NAT discovery port (default 4490)
      --natt-port uint16               IPSec NAT traversal port (default 4500)
      --password string                Password for basic authentication to the API server
      --proxy-url string               If provided, this URL will be used to connect via proxy
      --request-timeout string         The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
      --server string                  The address and port of the Kubernetes API server
      --tls-server-name string         If provided, this name will be used to validate server certificate. If this is not provided, hostname used to contact the server is used.
      --token string                   Bearer token for authentication to the API server
      --user string                    The name of the kubeconfig user to use
      --username string                Username for basic authentication to the API server
      --vxlan-port uint16              Internal VXLAN port (default 4800)
```

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

Closes #848

### CODE REVIEW

**Review by nyechiel — APPROVED** [review]

**Review by skitt — CHANGES_REQUESTED** [review]
My goal with https://github.com/submariner-io/submariner-website/pull/845 was to document the new connection options in a single location, so that we wouldn’t need to repeat them in every section. This *doesn’t* document the various `--as` etc. options in detail, but the reference for those is `kubectl`.

That way, we can concentrate on only the relevant options in each `subctl` command, instead of drowning them in the `kubectl`-provided options.

**Comment by dfarrell07** [comment]
> My goal with #845 was to document the new connection options in a single location, so that we wouldn’t need to repeat them in every section. This _doesn’t_ document the various `--as` etc. options in detail, but the reference for those is `kubectl`.
> 
> That way, we can concentrate on only the relevant options in each `subctl` command, instead of drowning them in the `kubectl`-provided options.

I added e1bffa6ecb3743970b45b1dd94fc5cccc1464d41 to address this, let me know if that's what you had in mind.

File: src/content/operations/deployment/subctl/_index.en.md [hunk 1]
```diff
@@ -397,26 +397,9 @@ This command prepares the underlying cloud infrastructure for Submariner install
...
 |:---------------------------------|:--------------------------------------------------------------------------------------------------------------------------|
-| `--as` `<string>`                       | Username to impersonate for the operation
-| `--as-group` `<stringArray>`            | Group to impersonate for the operation, this flag can be repeated to specify multiple groups
-| `--as-uid` `<string>`                   | UID to impersonate for the operation
-| `--certificate-authority` `<string>`    | Path to a cert file for the certificate authority
-| `--client-certificate` `<string>`       | Path to a client certificate file for TLS
-| `--client-key` `<string>`               | Path to a client key file for TLS
-| `--cluster` `<string>`                  | The name of the kubeconfig cluster to use
-| `--context` `<string>`                  | The name of the kubeconfig context to use
 | `--insecure-skip-tls-verify` `<string>` | If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
```
Comment (skitt) [inline src/content/operations/deployment/subctl/_index.en.md]: This is a connection flag too.

**Review by skitt — CHANGES_REQUESTED** [review]
Thanks, this is what I had in mind; there’s just one more flag to remove.

File: src/content/operations/deployment/subctl/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/operations/deployment/subctl/_index.en.md]: @sridhargaddam's original issue (https://github.com/submariner-io/submariner-website/issues/848) requested in particular `--insecure-skip-tls-verify` as a flag that should be added, so CCing him.

Comment (sridhargaddam) [reply src/content/operations/deployment/subctl/_index.en.md]: Right, the same flag is supported even for `subctl join ...` command as well. I'm fine if you want to include it in this PR or as a separate commit/PR. Thanks for looking into it @dfarrell07

**Comment by nyechiel** [comment]
@dfarrell07 are you looking into this?

**Comment by dfarrell07** [comment]
> @dfarrell07 are you looking into this?

@nyechiel I'm not sure what to do. As I understand the feedback, @skitt is saying the flag should not be documented here (instead, reference `subctl` output) and @sridhargaddam's original issue requests for it to be explicitly documented.

**Comment by dfarrell07** [comment]
> > @dfarrell07 are you looking into this?
> 
> @nyechiel I'm not sure what to do. As I understand the feedback, @skitt is saying the flag should not be documented here (instead, reference `subctl` output) and @sridhargaddam's original issue requests for it to be explicitly documented.

For now, I removed the common flag here and documented that decision on the original request issue (https://github.com/submariner-io/submariner-website/issues/848#issuecomment-1412487154).

**Review by skitt — APPROVED** [review]

**Pull request merged**